### PR TITLE
Implement no_std support

### PR DESCRIPTION
**Pull request by CryZe**
Turns out that supporting no_std is pretty trivial, so here are the changes necessary for that.

### CODE REVIEW

File: src/tables/aat.rs [hunk 1]
```diff
@@ -4,6 +4,8 @@ https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6AATIntro.h
 related types.
 */
 
+use core::cmp::Ordering;
```
Comment (RazrFalcon) [inline src/tables/aat.rs:7]: Prefer `core::cmp::Ordering::*`

File: src/tables/mod.rs [hunk 2]
```diff
@@ -1,3 +1,5 @@
+use core::{cmp::Ordering, fmt};
```
Comment (RazrFalcon) [inline src/tables/mod.rs:1]: Prefer `core::`

**Review by RazrFalcon — CHANGES_REQUESTED** [review]

File: src/buffer.rs [hunk 3]
```diff
@@ -1,4 +1,5 @@
-use std::convert::TryFrom;
+use alloc::{string::String, vec::Vec};
+use core::{convert::TryFrom, fmt, mem};
```
[the remark this answers is not in the document]
Comment (CryZe) [reply src/buffer.rs:2]: I'm not sure what you mean with that. You mean without the { } or what?

[the remark this answers is not in the document]
Comment (CryZe) [reply src/buffer.rs:2]: Oh you mean without importing the modules?

[the remark this answers is not in the document]
Comment (RazrFalcon) [reply src/buffer.rs:2]: Yes. Do not use `use` for `core`. `alloc` is fine.

**Comment by RazrFalcon** [comment]
Thanks.

File: src/buffer.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (CryZe) [reply src/buffer.rs]: Is my new push fine?

**Comment by CryZe** [comment]
Thank you as well :D

**Pull request merged**